controllers/autohealconfig_controller.go

```diff
@@ -47,16 +66,415 @@ type AutohealConfigReconciler struct {
+		if err != nil {
+			log.Errorf("Unable to get scenerios from config repo %s", err)
+
+		}
+		err = r.createConfigMaps(autoheal, scenarios, ctx)
+		if err != nil {
+			log.Errorf("Unable to create config maps %s", err)
+		}
+		err = r.removeUnwantedCM(ctx, &scenarios)
+		if err != nil {
+			log.Error("Unable to delete unwanted configs")
+		}
+	} else {
+		err = r.deleteConfigmap(autoheal, ctx)
+		if err != nil {
+			log.Errorf("Unable to reconcile deleting config maps %s", err)
+		}
 
-	// TODO(user): your logic here
+		err = r.deleteConfigSecret(autoheal, ctx)
+		if err != nil {
+			log.Errorf("Unable to reconcile deleting config secret %s", err)
+		}
+		if autoheal.Finalizers != nil {
+			autoheal.Finalizers = []string{}
+			err = r.Update(ctx, autoheal)
+			if err != nil {
+				log.Errorf("Unable to update finalizer to remove it %s", err)
+				return ctrl.Result{}, err
+			}
+		}
+	}
 
-	return ctrl.Result{}, nil
+	return r.updateStatus(autoheal)
 }
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *AutohealConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&autohealv1alpha1.AutohealConfig{}).
+		For(&v1alpha1.AutohealConfig{}).
 		Complete(r)
 }
+
+// Gets the secret needed to provide credentials to sycn with github repo
+func (r *AutohealConfigReconciler) GetConfigRepoSecret(ctx context.Context, autoheal *v1alpha1.AutohealConfig, status *v1alpha1.AutohealConfigStatus) (*v1alpha1.ConfigRepositoryInfo, error) {
+	log.Info("Autoheal-operator, Stage: Reading config secret")
+	configRepoSecret := &core.Secret{}
+	secretSelector := &client.ObjectKey{
+		Namespace: autoheal.Namespace,
+		Name:      AutohealConfigSecret,
+	}
+
+	err := r.Client.Get(ctx, *secretSelector, configRepoSecret)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Infof("Cannot find config repo secret %s", err)
+			return nil, nil
+		}
+		log.Errorf("Error cant get config repo secret %s", err)
+		return nil, err
+	}
+
+	if configRepoSecret.Name != AutohealConfigSecret {
+		log.Info("Config secret not found please create the autoheal config secret")
+		return nil, nil
+	}
+
+	configurl := string(configRepoSecret.Data["repository"])
+	_, err = url.Parse(configurl)
+	if err != nil {
+		log.Errorf("Unable to parse config URL from secret %s", err)
+		return nil, err
+	}
+
+	var configRepoInfo v1alpha1.ConfigRepositoryInfo
+	configRepoInfo.Tag = string(configRepoSecret.Data["tag"])
+	configRepoInfo.RepositoryURL = string(configRepoSecret.Data["repository"])
+	configRepoInfo.AccessToken = string(configRepoSecret.Data["access_token"])
+
+	status.LastSyncPeriod = time.Now().Unix()
+	return &configRepoInfo, nil
+}
+
+// Gets all the filename of the scripts from the repo
+func ReadFiles(repo *v1alpha1.ConfigRepositoryInfo) (*v1alpha1.AutohealConfigTree, error) {
+	log.Info(" Autoheal-operator Stage: Parsing file endpoints")
+	client := http.Client{}
```

Comment:
```suggestion
	httpClient := http.Client{}
```
My IDE is complaining that `Variable 'client' collides with imported package name`, same with occurrence in the `getScenarios` function.

Reply:
let me look into that

Reply:
i dont see that warning in my ide but ill change the name